### PR TITLE
[build] Fix rare failure case about dpkg frontend lock

### PR DESCRIPTION
**Pull request by liushilongbuaa**
#### Why I did it
There are some rare failure cases about dpkg frontend lock.
The conflict is:
1. Command 'dpkg -i' in docker-sonic-slave for target *-install
2. Command 'apt-get install' in docker build, such as target/docker-base-buster.gz.
#### How I did it
Add retry when running dpkg -i in docker-sonic-slave.
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012
- [x] 202106
- [x] 202111
- [x] 202205
- [x] 202211

### CODE REVIEW

File: slave.mk [hunk 1]
```diff
@@ -764,7 +764,14 @@ $(SONIC_INSTALL_DEBS) : $(DEBS_PATH)/%-install : .platform $$(addsuffix -install
...
 ifneq ($(CROSS_BUILD_ENVIRON),y)
-			{ sudo DEBIAN_FRONTEND=noninteractive dpkg -i $(DEBS_PATH)/$* $(LOG) && rm -d $(DEBS_PATH)/dpkg_lock && break; } || { set +e; rm -d $(DEBS_PATH)/dpkg_lock; sudo lsof /var/lib/dpkg/lock-frontend; ps aux; exit 1 ; }
+			# apt-get install inside sonic-slave may conflict with dpkg -i
+			# add retry here
+			for (( i=1; i<=10; i++))
+			do
+			    sudo DEBIAN_FRONTEND=noninteractive dpkg -i $(DEBS_PATH)/$* $(LOG) && rm -d $(DEBS_PATH)/dpkg_lock && break 2
+			    sleep 1
+			done
+			rm -d $(DEBS_PATH)/dpkg_lock; sudo lsof /var/lib/dpkg/lock-frontend; ps aux; exit 1
```
Comment (lguohan) [inline slave.mk:774]: why only fix the cross_build_environ, not the other if branch?

Comment (liushilongbuaa) [reply slave.mk:783]: fixed

File: slave.mk [hunk 2]
```diff
@@ -763,15 +763,24 @@ $(SONIC_INSTALL_DEBS) : $(DEBS_PATH)/%-install : .platform $$(addsuffix -install
 			{ while dpkg -s $(firstword $(subst _, ,$(basename $(deb)))) | grep "^Version: $(word 2, $(subst _, ,$(basename $(deb))))" &> /dev/null; do echo "waiting for $(deb) to be uninstalled" $(LOG); sleep 1; done } )
 		# put a lock here because dpkg does not allow installing packages in parallel
 		if mkdir $(DEBS_PATH)/dpkg_lock &> /dev/null; then
+
+		    # apt-get install inside sonic-slave may conflict with dpkg -i
+		    # add retry here
+		    for (( i=1; i<=10; i++))
```
Comment (xumia) [inline slave.mk:769]: Can we reduce the complex logic of the two loops?
```
while true; do
   for (( i=1; i<=10; i++))
   done
done
```

Comment (liushilongbuaa) [reply slave.mk:769]: The first one is to acquire dpkg lock.
The second one is to for dpkg install. We can't reduce the two loops.

Comment (liushilongbuaa) [reply slave.mk:769]: If reduce one loop, we can remove dpkg lock.
Retry is enough.

**Pull request closed without merging**